### PR TITLE
Do not show the connect button if admin config is not ok

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
For dashboard widget, sidebar tab, and personal setting:
- check if the admin config is ok (if all values are set and OAuth instance URL is valid)
- show connect button if ok
- show message `Failed to establish the connection. Please contact your administrator.` if not ok

### Related
https://community.openproject.org/projects/nextcloud-integration/work_packages/41211

### Screenshots
Screenshot from 2022-03-16 11-19-34

### CODE REVIEW

File: lib/Service/OpenProjectAPIService.php [hunk 1]
```diff
@@ -683,4 +683,16 @@ public function linkWorkPackageToFile(
...
+	/**
+	 * checks for every admin config variables if they are set
+	 *
+	 * @param IConfig $config
+	 * @return bool
+	 */
+	public static function isAdminConfigOk(IConfig $config):bool {
+		return !!($config->getAppValue(Application::APP_ID, 'client_id'))
+			&& !!($config->getAppValue(Application::APP_ID, 'client_secret'))
+			&& !!($config->getAppValue(Application::APP_ID, 'oauth_instance_url'));
```
Comment (individual-it) [inline lib/Service/OpenProjectAPIService.php:696]: should we also validate the URL here with `OpenProjectAPIService::validateOpenProjectURL`? or only when saving it?

File: src/components/tab/EmptyContent.vue [hunk 2]
```diff
@@ -7,7 +7,8 @@
 			<div class="empty-content--title">
 				{{ emptyContentMessage }}
 			</div>
-			<div v-if="showConnectButton" class="empty-content--connect-button">
+			<br>
+			<div v-if="showConnect && showConnectButton" class="connect-button">
```
Comment (individual-it) [inline src/components/tab/EmptyContent.vue:11]: I think this is strange naming `showConnect` and `showConnectButton`
what about doing it similar to https://github.com/nextcloud/integration_openproject/pull/73/files#diff-7f773d424a0ea444e1702d381d14bfdd2abcdbb0b199669aa28978cba4d46faeR104-R107 ?

**Review by individual-it — CHANGES_REQUESTED** [review]
I think we also need an extra message for the user if the setting is not complete.
e.g. "OpenProject app has not been configured correctly, please contact the NextCloud administrator"

File: lib/Service/OpenProjectAPIService.php (hunk 1 above)
Comment (kiranparajuli589) [reply lib/Service/OpenProjectAPIService.php]: OAuth instance URL should also be valid for a successful connection. Added a check for URL too.

File: src/components/tab/EmptyContent.vue (hunk 2 above)
Comment (kiranparajuli589) [reply src/components/tab/EmptyContent.vue]: fixed

File: src/components/OAuthConnectButton.vue [hunk 3]
```diff
@@ -20,7 +24,18 @@ export default {
...
 		},
+		adminConfigStatus: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	computed: {
+		adminConfigNotOkMessage() {
+			return t('integration_openproject', 'Failed to establish connection. Please contact your administrator.')
```
Comment (individual-it) [inline src/components/OAuthConnectButton.vue:35]:
```suggestion
			return t('integration_openproject', 'Settings of the OpenProject integration app are not complete. Please contact your NextCloud administrator.')
```

**Review by individual-it — CHANGES_REQUESTED** [review]
just a little suggestion to change the text

**Review by individual-it — APPROVED** [review]

**Pull request merged**